GVRf/Framework/jni/engine/renderer/renderer.cpp

```diff
@@ -504,66 +500,83 @@ void Renderer::renderRenderData(RenderData* render_data,
+                    glm::mat4 mvp_matrix(projection_matrix * mv_matrix);
+                    try {
+                        bool right = render_mask & RenderData::RenderMaskBit::Right;
+                        switch (curr_material->shader_type()) {
+                        case Material::ShaderType::UNLIT_SHADER:
+                            shader_manager->getUnlitShader()->render(mvp_matrix,
+                                    render_data, curr_material);
+                            break;
+                        case Material::ShaderType::UNLIT_HORIZONTAL_STEREO_SHADER:
+                            shader_manager->getUnlitHorizontalStereoShader()->render(
+                                    mvp_matrix, render_data, curr_material,
+                                    right);
+                            break;
+                        case Material::ShaderType::UNLIT_VERTICAL_STEREO_SHADER:
+                            shader_manager->getUnlitVerticalStereoShader()->render(
+                                    mvp_matrix, render_data, curr_material,
+                                    right);
+                            break;
+                        case Material::ShaderType::OES_SHADER:
+                            shader_manager->getOESShader()->render(mvp_matrix,
+                                    render_data, curr_material);
+                            break;
+                        case Material::ShaderType::OES_HORIZONTAL_STEREO_SHADER:
+                            shader_manager->getOESHorizontalStereoShader()->render(
+                                    mvp_matrix, render_data, curr_material,
+                                    right);
+                            break;
+                        case Material::ShaderType::OES_VERTICAL_STEREO_SHADER:
+                            shader_manager->getOESVerticalStereoShader()->render(
+                                    mvp_matrix, render_data, curr_material,
+                                    right);
+                            break;
+                        case Material::ShaderType::CUBEMAP_SHADER:
+                            shader_manager->getCubemapShader()->render(
+                                    model_matrix, mvp_matrix, render_data,
+                                    curr_material);
+                            break;
+                        case Material::ShaderType::CUBEMAP_REFLECTION_SHADER:
+                            shader_manager->getCubemapReflectionShader()->render(
+                                    mv_matrix, glm::inverseTranspose(mv_matrix),
+                                    glm::inverse(view_matrix), mvp_matrix,
+                                    render_data, curr_material);
+                            break;
+                        default:
+                            shader_manager->getCustomShader(
+                                    curr_material->shader_type())->render(
```

Comment:
Phil, to reproduce it, leave the app run for a while, say for more than 10 minutes. It can be problem of my test device, but is consistent on my Note 4.